vault/auth.go

```diff
@@ -148,6 +155,63 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	if env.Getenv("VAULT_AUTH_AWS_METHOD", "ec2") != "ec2" {
+		return ""
+	}
+
+	if skip := env.Getenv("VAULT_AUTH_AWS_EC2_SKIP", ""); skip != "" {
```

Comment:
Isn't `VAULT_AUTH_AWS_METHOD` required to differentiate between `ec2` and `iam`? Or do you mean you'd default to `ec2`?